src/main/java/org/folio/consortia/service/impl/PublicationServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package org.folio.consortia.service.impl;
+
+import static org.folio.consortia.utils.TenantContextUtils.prepareContextForTenant;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.consortia.domain.dto.PublicationRequest;
+import org.folio.consortia.domain.dto.PublicationResponse;
+import org.folio.consortia.exception.PublicationException;
+import org.folio.consortia.service.HttpRequestService;
+import org.folio.consortia.service.PublicationService;
+import org.folio.consortia.service.TenantService;
+import org.folio.consortia.service.UserTenantService;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.FolioModuleMetadata;
+import org.folio.spring.scope.FolioExecutionContextSetter;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@AllArgsConstructor
+public class PublicationServiceImpl implements PublicationService {
+  private final TenantService tenantService;
+  private final UserTenantService userTenantService;
+  private final FolioExecutionContext folioExecutionContext;
+  private final FolioModuleMetadata folioModuleMetadata;
+  private final HttpRequestService httpRequestService;
+  private final TaskExecutor asyncTaskExecutor;
```

Comment:
Is this task executor limited to a separate thread pool? Hoping that all threads for the module are not exhausted by multiple running publications.